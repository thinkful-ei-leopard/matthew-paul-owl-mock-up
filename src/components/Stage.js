import React from "react";
import StageParticipant from "./StageParticipant";

function Stage(props) {
  return (
    <div className="stage">
      {props.participants.map(person => (
        <StageParticipant
          key={person.id}
          name={person.name}
          avatar={person.avatar}
        />
      ))}
    </div>
  );
}

export default Stage;
