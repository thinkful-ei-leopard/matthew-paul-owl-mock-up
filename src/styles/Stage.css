.Stage {
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    align-items: center;
    background: #1E2835;
    width: 100%;
    padding: 20px;
}

.Stage-Participant {
    text-align: center;
    padding: 10px;
}

.Participant-Info-Container {
    color: white;
    background: #2A3340;
    padding: 15px;
    border-radius: 10px;
}

.Stage-Avatar {
    padding: 5px;
    border-radius: 50%;
}