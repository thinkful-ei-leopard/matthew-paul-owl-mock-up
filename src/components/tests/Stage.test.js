import React from 'react';
import ReactDOM from 'react-dom';
import Stage from '../Stage';
import renderer from 'react-test-renderer';
import STORE from '../../STORE';

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<Stage participants={STORE.participants}/>, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
  .create(<Stage participants={STORE.participants}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });
