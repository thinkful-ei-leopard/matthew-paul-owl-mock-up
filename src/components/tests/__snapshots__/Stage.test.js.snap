// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="stage"
>
  <div
    className="stage-participant"
  >
    <div
      className="participant-info-container"
    >
      <span
        className="participant-name"
      >
        Koren Templeton
      </span>
    </div>
    <img
      alt="avatar"
      className="stage-avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="stage-participant"
  >
    <div
      className="participant-info-container"
    >
      <span
        className="participant-name"
      >
        Caty Flucker
      </span>
    </div>
    <img
      alt="avatar"
      className="stage-avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="stage-participant"
  >
    <div
      className="participant-info-container"
    >
      <span
        className="participant-name"
      >
        Axe Kubicka
      </span>
    </div>
    <img
      alt="avatar"
      className="stage-avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="stage-participant"
  >
    <div
      className="participant-info-container"
    >
      <span
        className="participant-name"
      >
        Frank Runciman
      </span>
    </div>
    <img
      alt="avatar"
      className="stage-avatar"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="stage-participant"
  >
    <div
      className="participant-info-container"
    >
      <span
        className="participant-name"
      >
        Ashla Attwool
      </span>
    </div>
    <img
      alt="avatar"
      className="stage-avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
  </div>
</div>
`;
